CMakeLists.txt

```diff
@@ -9,9 +9,18 @@ message(STATUS "Using CMake version: " ${CMAKE_VERSION})
 # https://github.com/pybind/pybind11/issues/4825
 set(CMAKE_INTERPROCEDURAL_OPTIMIZATION OFF)
 
+# Default to not using OpenMP unless explicitly requested since it can introduce some issues
```

Comment:
Can you remind me which issues these are? Is it the question about the OpenMP compiler flags on OSX?

Reply:
The main issue is that the compile will fail if OpenMP isn't available since FINUFFT defaults to requiring OpenMP. You face this a lot on Mac because it's a PITA to install OpenMP. We could potentially change this to only default to `OFF` on Mac, but I thought it might be better to just be explicit everywhere. I don't feel strongly!

Reply:
I would lean towards leaving the default being OpenMP enabled on Linux, just so the default is the fastest behavior (and since that's what we've been doing up until now). But I likewise don't feel strongly, there are pros and cons to both approaches!

Reply:
OK good point. I've updated this to only enable OpenMP only if `find_package(OpenMP)` succeeds, but this can be disabled with `JAX_FINUFFT_USE_OPENMP=OFF`. This seems to work on both Mac and Linux as expected.